packages/admin/src/form/FinalFormMultiSelect.tsx

```diff
@@ -0,0 +1,95 @@
+import { Check } from "@comet/admin-icons";
+import { createStyles, List, ListItem, ListItemIcon, ListItemText, WithStyles, withStyles } from "@material-ui/core";
+import * as React from "react";
+import { FieldRenderProps } from "react-final-form";
+
+/*---------------- FinalFormMultiSelectListItemComponent ----------------*/
+
+interface FinalFormMultiSelectListDefaultItemProps {
+    option: Option;
+    isSelected: boolean;
+}
+
+function FinalFormMultiSelectListDefaultItem({ option, isSelected }: FinalFormMultiSelectListDefaultItemProps) {
+    return (
+        <>
+            {option.icon && <ListItemIcon>{option.icon}</ListItemIcon>}
+            <ListItemText>{option.label}</ListItemText>
+            {isSelected && <Check />}
+        </>
+    );
+}
+
+/*---------------- FinalFormMultiSelectComponent ----------------*/
+
+export interface Option {
+    label: string | React.ReactNode;
+    value: string;
+    icon?: React.ReactNode;
+}
+
+interface FinalFormMultiSelectProps extends FieldRenderProps<string, HTMLDivElement> {
+    options: Option[];
+    renderItem?: (option: Option) => React.ReactNode;
+}
+
+export type FinalFormMultiSelectClassKey = "root" | "listItem" | "listItemSelected";
+const styles = () => {
+    return createStyles<FinalFormMultiSelectClassKey, FinalFormMultiSelectProps>({
+        root: {},
+        listItem: {},
+        listItemSelected: {},
+    });
+};
+
+function FinalFormMultiSelectComponent({ options, renderItem, input, classes }: WithStyles<typeof styles> & FinalFormMultiSelectProps) {
+    const handleListItemClick = (value: string) => {
+        if (Array.isArray(input.value)) {
```

Comment:
Shouldn't this value always be an array? As in _multi_ select?

Reply:
on the first render if nothing is selected the value of the input field is null/undefined from final form i guess thats why i had to check it first before i can use the .includes() method on the input.value :)

Reply:
if input can be null/undefined don't use isArray on it. When reading the code that can be confusing, as you think non-array values are possible.

`if (!Array.isArray(....)) throw new Error(....);` might still be useful.

And using `[undefined]` as you do in the else cause seems not correct.

Reply:
it is usually always an array but every final form field is initially undefined thats why i have to check if already a value is set so i can do the check if a value got already clicked or need to be added/removed. 

And yes its true with that solution a [undefined] would have been possible :) Now i only call onChange of the input if the value is !== undefined :)

Reply:
Do I understand correctly, there are to cases for value:

- undefined
- Array

if that is true, you only have to handle those two cases

Reply:
In the first render if the field gets no value from initialValues it has the value undefined. If now i choose some values i have to create an array with those values. So if i would not do the check Array.isArray on the first selection of values i would have a problem if i then use .includes() on the value because its undefined. I tried to make the selection as save as possible to avoid any kind of errors.

Reply:
When I read `Array.isArray(input.value)`, I think this can be more than Array | undefined, which isn't true tough.

Better: `input.value !== undefined`